client/consensus/src/collators/lookahead.rs

```diff
@@ -342,6 +613,34 @@ where
                             Ok(h) => h,
                         };
 
+                        // TODO: Currently we use just the first core here, but for elastic scaling
+                        // we iterate and build on all of the cores returned.
+                        // More info: https://github.com/paritytech/polkadot-sdk/issues/1829
+                        let (is_parachain, core_index) = match (&aux_data.slot_freq, core_indices.first()) {
+                            (None, None) => break, // We are parachain and we do not have core allocated, nothing to do,
+                            (None, Some(core_index)) => (true, core_index), // We are parachain and we have core allocated, let's continue
```

Comment:
Love these comments, maybe we can turn them into `log::trace!` logs? So when we are trying to debug why a collator is not producing blocks, we can enable that log target.

Reply:
Sure. Will do that.